pkg/buildpack/multi_architecture_helper.go

```diff
@@ -0,0 +1,170 @@
+package buildpack
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/pack/internal/paths"
+	"github.com/buildpacks/pack/internal/style"
+	"github.com/buildpacks/pack/pkg/dist"
+	"github.com/buildpacks/pack/pkg/logging"
+)
+
+type MultiArchConfig struct {
+	buildpackTargets []dist.Target
+	expectedTargets  []dist.Target
+	logger           logging.Logger
+}
+
+func NewMultiArchConfig(targets []dist.Target, expected []dist.Target, logger logging.Logger) (MultiArchConfig, error) {
+	// Let's do some validations
+	return MultiArchConfig{
+		buildpackTargets: targets,
+		expectedTargets:  expected,
+		logger:           logger,
+	}, nil
+}
+
+func (m *MultiArchConfig) Targets() []dist.Target {
+	if len(m.expectedTargets) == 0 {
+		return m.buildpackTargets
+	}
+	return m.expectedTargets
+}
+
+// CopyConfigFiles Given a base directory, which is expected to be the root folder of single buildpack,
+// it will copy a buildpack.toml file, for each target, into the corresponding platform root folder.
+func (m *MultiArchConfig) CopyConfigFiles(baseDir string) ([]string, error) {
+	var filesToClean []string
+	for _, target := range m.Targets() {
+		// TODO we are not handling distributions versions yet
+		path, err := CopyConfigFile(baseDir, target, "")
+		if err != nil {
+			return nil, err
+		}
+		if path != "" {
+			filesToClean = append(filesToClean, path)
+		}
+	}
+	return filesToClean, nil
+}
+
+// PrepareDependencyConfigFile when creating a composite buildpack, dependencies URI are relative to the main buildpack
+// if we are building a multi-arch composite buildpack we MAY need to copy the buildpack.toml file to the dependency and
+// determine the platform root folder. This method will do those operations and return the path to the buildpack.toml that
+// was copied.
```

Comment:
I'm still confused as to what this function is doing and why. Maybe resolving the below questions will help...

Reply:
Does this [comment](https://github.com/buildpacks/pack/pull/2086/commits/02b7efc3b7221ddcefd8e02d8fce8e6a8b714e26#r1591516026) help?

Reply:
I went through the code again, this code was for handling the case I mentioned in my previous comment. 

When we try to build a multi-platform composite buildpack and in the `dependencies.URI` we point to a file path, but because we simplified this logic requesting to push the dependencies to a registry first, then I just removed this code.